ui/app/adapters/kv/data.js

```diff
@@ -31,6 +38,19 @@ export default class KvDataAdapter extends ApplicationAdapter {
     });
   }
 
+  // TODO use query-param-string util when https://github.com/hashicorp/vault/pull/27455 is merged
+  fetchSubkeys(query) {
+    const { backend, path, version, depth } = query;
+    const apiPath = buildKvPath(backend, path, 'subkeys'); // encodes mount and secret paths
```

Comment:
Any reason you didn't export a `kvSubkeyPath` method instead of using the base helper? Then the version and query params logic could be in there as well, and easy to unit test

Reply:
Since this endpoint isn't implemented yet, it was easier for my brain to fiddle with and update the logic in one place. I can move this into a `kvSubkeyPath` helper. The adapter unit test felt sufficient for testing, but I can add another test for the util.